src/http_server.cc

```diff
@@ -4489,13 +4489,13 @@ HTTPAPIServer::GenerateRequestClass::ExactMappingOutput(
           *generate_response, triton::common::TritonJson::ValueType::ARRAY);
       RETURN_IF_ERR(WriteDataToJson(
           &data_json, cname, datatype, base, byte_size, element_count));
-      if (element_count > 1) {
-        RETURN_IF_ERR(generate_response->Add(cname, std::move(data_json)));
-      } else {
+      if (element_count == 1) {
         // if only 1 element, strip out the array
         triton::common::TritonJson::Value el;
         RETURN_IF_ERR(data_json.At(0, &el));
         RETURN_IF_ERR(generate_response->Add(cname, std::move(el)));
+      } else {
```

Comment:
Was the wrong case here `element_count == 0` ?

Reply:
Yes. The error `attempt to access non-existing array index '0'` occurred before this fix was because `data_json` was an empty array and this line  `RETURN_IF_ERR(data_json.At(0, &el));` was called for `element_count == 0`.

Reply:
Can you share the full output from generate endpoint for one of these zero dim tensors after the fix?

Reply:
Both `/generate` and `/generate_stream` *

Reply:
Sure! Added the output in the PR description.

Reply:
Thanks!